internal/agent/device/device.go

```diff
@@ -383,10 +387,16 @@ func (a *Agent) systemdControllerSync(_ context.Context, desired *v1alpha1.Rende
 	return nil
 }
 
-func (a *Agent) afterUpdate(ctx context.Context) error {
+func (a *Agent) afterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
 	a.log.Debug("Executing post actions")
 	defer a.log.Debug("Finished executing post actions")
 
+	rebooted := false
+	if err := a.hookManager.OnAfterUpdating(ctx, current, desired, rebooted); err != nil {
```

Comment:
I believe we want this at the end of afterUpdate(). For example if ExecuteActions fails do we want these hooks to fire?

Reply:
Importantly, we need to explicit for each hook about what happens when an action fails (see PR #619). I believe that, yes, also for an afterUpdating hook we should by default roll back when an action fails. Let's say an invalid firewall rule that fails to activate.

There may be other actions that are less critical. In that case, maybe the user should be able to explicitly specify what to do, e.g. adding an `onError:` field to the action that can `rollback` or `ignore`.